### ISSUE TITLE
Add optional Config param upload_path to config.xml or use defaults

### ISSUE THREAD

**Issue by DanielSchwiperich** [contao/standard-edition#26]
When using the standard editon the `config.yml` is missing at least one important _optional_ parameter:
**upload_path** 

The files folder is created via `InstallCommand::addEmptyDirs()` from the core bundle but it gets not symlinked in `SymlinksCommand::symlinkFiles` if the config param `upload_path` is missing.

So either the param should not be optional and being provided in the `config.yml` or a default / fallback value should be used in the `SymlinksCommand::generateSymlinks()` method of the core bundle where currently it is not:

```
$uploadPath = $this->getContainer()->getParameter('contao.upload_path');

// Remove the base folders in the document root
        $fs->remove($this->rootDir . '/web/' . $uploadPath);
        $fs->remove($this->rootDir . '/web/system/modules');
        $fs->remove($this->rootDir . '/web/vendor');

$this->symlinkFiles($uploadPath);
```

https://github.com/contao/core-bundle/blob/master/src/Command/SymlinksCommand.php#L68

I think this could also be considered a general bug (core-bundle related)  if `the upload_path` is empty because for backend uploads still the `files` - folder is used as configured in the `default.php` 
https://github.com/contao/core-bundle/blob/master/src/Resources/contao/config/default.php#L80

**Comment by aschempp** [comment]
The upload path is set by the default configuration of the `contao/core-bundle`. Are you sure it's missing?
https://github.com/contao/core-bundle/blob/master/src/DependencyInjection/Configuration.php#L45-L57

**Comment by DanielSchwiperich** [comment]
Thanks for clarification @aschempp and you're right the value is not missing. It was my assumption.

Just found **I** was missing the new workflow that you now have to explicit publish folders, instead of protecting them like in 3.5 :smile: 

Still I think there should be a constant defined for such a default path / value instead of reusing strings _all over the place_

**Comment by aschempp** [comment]
No problem. What do you mean by "reusing strings"? Constants are bad practice, that's exactly what the container configuration is for.

**Comment by DanielSchwiperich** [comment]
Yes a constant would be not that fine, you're right (again :) )
I came to my assumption while studying the InstallCommand.php and SymlinksCommand.php
I saw that empty dirs there were created and some symlinks for the web dir. 

In the empty dirs array it uses 'files' and since I could not access my uploaded files I was assuming the cause is, it gets not symlinked because upload_path is missing. 

Would the `InstallCommand::addEmptyDirs()` use the config I would have stumbled about the Configuration Class where the default are initialized. 

But this leads to a real bug which core-bundle related. 
Will open it there.
Thanks again
